frontend/src/components/Creators.jsx

```diff
@@ -1,109 +1,68 @@
-    this.setState({ creators: this.props.creators });
+    this.handleFilterChange = this.handleFilterChange.bind(this);
   }
 
-  onChange(ev) {
-    console.log(ev.target.name, ev.target.value);
-
-    const { value } = ev.target;
+  handleFilterChange(name, value) {
     this.setState({
-      [ev.target.name]: value
+      [name]: value
     });
   }
 
-  onSubmit() {
-    ev.preventDefault();
-
-    const { searchWords, state, gender } = this.state;
-
-    const creators =
-      state.length === 0
-        ? this.creators.filter(
-            creator => creator.location.split(", ")[1] === state
-          )
-        : this.creators;
-
-    if (gender.length === 0) {
-      const filterByGender = creators.filter(
-        creator => creator.gender === gender
-      );
-      creators = filterByGender;
-    }
-
-    if (searchWords.length !== 0) {
-      const search = searchWords.toLowerCase();
-      const searchObj = ["first_name", "last_name", "description", "interest"];
-      // need to check what to filter by the search key word
-    }
-
-    this.setState({ searchWords: "", state: "", gender: "", creators });
-  }
-  */
   render() {
-    const { creators } = this.props;
-    const { onChange, onSubmit } = this;
-
+    const { creators, locations } = this.props;
+    const { searchText, searchLocation, searchGender } = this.state;
+    const { handleFilterClear, handleFilterChange } = this;
     return (
       <div>
-        <form>
-          <input
-            type="search"
-            name="searchWords"
-            placeholder="Search by keywords"
-            onChange={onChange}
-          ></input>
-          <select name="state">
-            {/* {creators.map(creator => {
-              const state = creator.location.name.split(", ")[1];
-              return <option value={state}>{state}</option>;
-            })} */}
-          </select>
-          <select name="gender">
-            <option value="female">female</option>
-            <option value="male">male</option>
-          </select>
-          <button type="submit" onSubmit={onSubmit}>
-            Update
-          </button>
-        </form>
-        <CreatorList>
-          {creators.map(creator => (
-            <CreatorCard key={creator.id} creator={creator} />
-          ))}
-        </CreatorList>
+        <CreatorSearchBar
+          searchText={searchText}
+          searchLocation={searchLocation}
+          searchGender={searchGender}
+          locations={locations}
+          onFilterChange={handleFilterChange}
+        />
+        <CreatorList
+          creators={creators}
+          searchText={searchText}
+          searchLocation={searchLocation}
+          searchGender={searchGender}
+          locations={locations}
+        />
       </div>
     );
   }
 }
 
-const mapStateToProps = ({ creators }) => ({ creators });
+const mapStateToProps = ({ creators }) => {
+  const locations =
+    creators &&
+    creators.reduce((accum, ele) => {
```

Comment:
I got an error when I trying to search a creator, I type in a letter and hit return, I got an error says "Creators.jsx:67 Uncaught TypeError: creators.reduce is not a function"

Reply:
This t happens when the page is refreshed since the redux store states lost after reload, but filter and creator pages are working. We may need to revise app mount to resolve this issues.